services/site/src/app/challenge/[challengeSlug]/page.tsx

```diff
@@ -0,0 +1,110 @@
+import { getDifficultyIconByChallengeDifficulty } from "app/components/challengePage/difficulties/Difficulties";
+import Footer from "app/components/Footer";
+import Check from "app/components/icons/Check";
+import Navigation from "app/components/Navigation";
+import { ChallengeDetailsBySlugDocument, ChallengeDetailsBySlugQuery, ChallengeDetailsBySlugQueryResult, LevelStatus } from "app/generated/graphql";
+import { getApolloClient } from "app/lib/apollo-client/rsc";
+import clsx from "clsx";
+import { Metadata } from "next";
+import Link from "next/link";
+import { notFound } from "next/navigation";
+import sanitizeHtml from "sanitize-html";
+
+type PageProps = {
+  params: {
+    challengeSlug: string;
+  };
+};
+
+async function getChallenge(slug: string): Promise<ChallengeDetailsBySlugQueryResult["data"]["challenge"]> {
+  const client = getApolloClient();
+
+  const response = await client.query<ChallengeDetailsBySlugQuery>({
+    query: ChallengeDetailsBySlugDocument,
+    variables: { slug },
+  });
+
+  return response.data.challenge;
+}
+
+const Challenge = async ({ params: { challengeSlug } }: PageProps): Promise<React.ReactElement> => {
+  const challenge = await getChallenge(challengeSlug);
+
+  if (!challenge) {
+    notFound();
+  }
+
+  const DifficultyIcon = getDifficultyIconByChallengeDifficulty(challenge.difficulty);
+  const firstUnfinishedLevel = challenge.levels.find((level) => level.status === LevelStatus.Open || level.status === LevelStatus.InProgress);
+
+  return (
+    <>
+      <Navigation displayBreadcrumbs />
+      <main className="h-full box-border max-w-screen-3xl mx-auto">
+        <div className="mx-8 py-8 h-main max-w-screen-3xl sm:mx-12 lg:mt-12 lg:mx-24">
+          <h1 className="mb-8 h2 md:h1">
+            <span className="sr-only">Challenge: </span>
+            {challenge?.name}
+            <div className={clsx("inline-block")}>
+              <DifficultyIcon className={clsx("w-5 h-12 inline-block")} firstClassName="ml-4" />
```

Comment:
We miss an accessible name here. Within the component we should have a sr-only p that says "Difficulty Easy" for example. Then it is like it is in the Challenge car that gets displayed on the homepage. Maybe we need to add it to the component and remove it from the challenge card. Depends on how the code currently looks.